code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -327,7 +327,10 @@
 
 					visible_message("<span class='danger'>[src] pounces on [M]!</span>",
 									"<span class='xenodanger'>We pounce on [M]!</span>", null, 5)
-					M.Paralyze(20)
+					if(xeno_caste.charge_type == CHARGE_TYPE_PANTHER)
```

Comment:
please dont do this, this code is already pretty bad.

Reply:
What should I replace it with? This is my first time coding. Would M.Paralyze((xeno_caste.charge_type == CHARGE_TYPE_PANTHER) ? 45 : 20) be just as bad?

Reply:
special casing the charge type inside a ranged switch case to just change the stun timer is not great.

Reply:
ideally this would be determined by the pounce action and not hard coded like this

Reply:
i can do this for you since you're new

Reply:
You know what this is fine as-is i'll just merge this and pr a refactor of this whole messy proc

Reply:
Thank you